### PR TITLE
Issue #2068: updated RedundantModifier documentation for final in enums

### PR DESCRIPTION
**Pull request by rnveach**
Issue #2068

Updated documentation and added test to show.

### CODE REVIEW

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/modifier/InputFinalInEnumMethods.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.puppycrawl.tools.checkstyle.checks.modifier;
+
+public enum InputFinalInEnumMethods {
+    E1,
+    E2 {
+        @Override
+        public final void v() {
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/modifier/InputFinalInEnumMethods.java]: why this "final" is not redundant ? Can we extend E2 ?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.java [hunk 2]
```diff
@@ -54,6 +54,9 @@
  * <p>Enums by definition are static implicit subclasses of java.lang.Enum&#60;E&#62;.
  * So, the <code>static</code> modifier on the enums is redundant. In addition,
  * if enum is inside of interface, <code>public</code> modifier is also redundant.
+ * Enums can contain abstract methods which can be overriden by the declared fields.
+ * Since they can be overridden, final methods in these types are not marked as redundant
+ * even though they can't be extended by other classes/enums.
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.java]: I failed to understand this .... till I read examples in issue. We need to think about this description one more time.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/modifier/InputFinalInEnumMethods.java (hunk 1 above)
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/modifier/InputFinalInEnumMethods.java]: ignore this comment , I did not noticed #3582

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.java (hunk 2 above)
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.java]: > Enums can contain abstract methods

not only abstract, as we see in example.

Please provide an example, and by trailing comment show where violation be and where not.

Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheck.java]: Done. Wording updated and added example.

**Pull request merged**